### PR TITLE
add(): add CRD for AzureSecretEngineRole

### PR DESCRIPTION
**Pull request by erlisb**
This PR consists in :

- API Resource / CRD for AzureSecretEngineRole

- Defaulter / Validator Webhook for AzureSecretEngineRole CRD

- Controller Reconciliator for AzureSecretEngineRole CRD

- Documentation for AzureSecretEngineRole CRD

@raffaelespazzoli , please review it.

Thnx

### CODE REVIEW

**Comment by raffaelespazzoli** [comment]
@erlisb could you resolve the conflicts? Then I can merge

**Comment by erlisb** [comment]
Done.

**Comment by raffaelespazzoli** [comment]
@erlisb can you fix the conflict?

**Comment by raffaelespazzoli** [comment]
@erlisb ping...

**Comment by erlisb** [comment]
a bit of pain, but please, try again :)

File: PROJECT [hunk 1]
```diff
@@ -424,5 +424,18 @@ resources:
...
+    namespaced: true
+  controller: true
+  domain: redhat.io
+  group: redhatcop
+  kind: AzureSecretEngineRole
+  path: github.com/redhat-cop/vault-config-operator/api/v1alpha1
+  version: v1alpha1
+  webhooks:
+    defaulting: true
+    validation: true.
```
Comment (raffaelespazzoli) [inline PROJECT:439]: please fix this. It's causing an error at build time

**Review by raffaelespazzoli — CHANGES_REQUESTED** [review]
@erlisb

File: PROJECT (hunk 1 above)
Comment (erlisb) [reply PROJECT]: Done

**Comment by raffaelespazzoli** [comment]
@erlisb the `VaultObject` interface was modified by a previous PR, you now have to implement the `IsDeletable` method. For this CR the right answer is true.
If you open the code with an IDE, you should clearly see the issue.

**Comment by erlisb** [comment]
Hi @raffaelespazzoli I added the missing method, as suggested. 
Can you please, check again ? Thnx

**Comment by erlisb** [comment]
Hi @raffaelespazzoli kind reminder!

**Pull request merged**